### ISSUE TITLE
[WebAssembly] ASSERTION 2257: ChakraCore v1.11.7

### ISSUE THREAD

**Issue by kjpou1** [Microsoft/ChakraCore#6037]
Hello

I have validated that the test case for https://github.com/Microsoft/ChakraCore/issues/5985 runs correctly with a Debug version of ChakraCore and the small test case but there still seems to be something wrong when running against our tests and still receiving the following:

```
ASSERTION 2257: (/Users/Jimmy/websharp/projects/ChakraCore/lib/Backend/IRBuilderAsmJs.cpp, line 1044) !IsLoopBody() || offset <= GetLoopBodyExitInstrOffset()
 Failure: (!IsLoopBody() || offset <= GetLoopBodyExitInstrOffset())
Illegal instruction: 4
```

Not sure how to create a small repo of this either.  Is there some way to help narrow this down even further?

**Comment by kjpou1** [comment]
Chakra version https://github.com/Microsoft/ChakraCore/commit/e6e25c3a5ec0f4d2bda13eb6f9b06550d91daddb

**Comment by MikeHolman** [comment]
Do you have any crash dumps or anything like that? Can you pass custom switches to chakracore? If so, `-bgjit- -lic:1` might make repro more consistent.

Do you have a larger repro which I can run using ch?

**Comment by kjpou1** [comment]
I will try the custom switch.  Also, will try to get a larger repro that you can run using ch.

**Comment by kjpou1** [comment]
Hello Mike

I ran the tests using Charka debug build version:  https://github.com/Microsoft/ChakraCore/commit/ca582450de28ef6179e9648030190f0eaa7eea74

The test file can be found [here:](https://microsoft-my.sharepoint.com/:u:/p/kepounce/EfEoBZng4JVGmmn1fGbhZWEBVsmkvogkJm04hxfIUjxlEQ?e=7FCxIn)

If you can not access it please let me know.  Should be accessible by everyone.

The output as follows:

```

/Users/Jimmy/websharp/projects/ChakraCore/out/Debug/bin/ch/ch runtime.js -args --run hello.exe
Arguments: --run,hello.exe
Loaded: hello.exe
Loaded: mscorlib.dll
initializing mono runtime
>>mono_wasm_runtime_ready
WASM: Initializing.....
ASSERTION 12765: (/Users/Jimmy/websharp/projects/ChakraCore/lib/Backend/IRBuilderAsmJs.cpp, line 1044) !IsLoopBody() || offset <= GetLoopBodyExitInstrOffset()
 Failure: (!IsLoopBody() || offset <= GetLoopBodyExitInstrOffset())
Illegal instruction: 4

```

Run with `-bgjit- -lic:1` option as described above.

```
/Users/Jimmy/websharp/projects/ChakraCore/out/Debug/bin/ch/ch -bgjit- -lic:1 runtime.js -args --run hello.exe
Arguments: --run,hello.exe
Loaded: hello.exe
Loaded: mscorlib.dll
initializing mono runtime
ASSERTION 12772: (/Users/Jimmy/websharp/projects/ChakraCore/lib/Backend/IRBuilderAsmJs.cpp, line 1044) !IsLoopBody() || offset <= GetLoopBodyExitInstrOffset()
 Failure: (!IsLoopBody() || offset <= GetLoopBodyExitInstrOffset())
Illegal instruction: 4

```

Let me preface the above two runs saying that it sometimes passes.  I have gone up to 20 calls before it fails with the errors above. 

Here is the output from a successful run:

```
/Users/Jimmy/websharp/projects/ChakraCore/out/Debug/bin/ch/ch runtime.js -args --run hello.exe
Arguments: --run,hello.exe
Loaded: hello.exe
Loaded: mscorlib.dll
initializing mono runtime
>>mono_wasm_runtime_ready
WASM: Initializing.....
WASM: ﻿Hello, World!
```

**Comment by kjpou1** [comment]
@lewing ^

**Comment by renatahodovan** [comment]
I'm not a ChakraCore developer, but tried to download the test case and could not access it.

**Comment by kjpou1** [comment]
Thank you for letting me know @renatahodovan.

Try this one: https://www.dropbox.com/s/l4h3uw0cdvb9st1/interp-sample.zip?dl=0

**Comment by kjpou1** [comment]
@MikeHolman So far using chakra build from chakra repo: https://github.com/Microsoft/ChakraCore with the fix Microsoft/ChakraCore#6072 applied all the tests are running through. Have executed tests locally with the build multiple times.

Thanks so much.

Do you guys know when a new version will be out so we can switch our CI tests back on for Chakra?

**Comment by renatahodovan** [comment]
It's probably a late hit, but I played with various automatic reduction approaches[1][2] applied on the provided test case and got the attached results[3]. I've posted the reduced version in case you need a regression test for the fix.

[1] https://github.com/renatahodovan/picire
[2] https://github.com/renatahodovan/picireny
[3] Results:
    mono.js    : 497995 B   -> 1592 B
    mono.wasm  : 1684878 B  -> 1782 B
    mono.wast  : 25748578 B -> 5512 B
    runtime.js : 9243 B     -> 234 B
    mono-config.js got manually inlined where it's used
    the managed directory is not needed to reproduce the issue

[minimal.zip](https://github.com/Microsoft/ChakraCore/files/3068699/minimal.zip)

**Comment by MikeHolman** [comment]
The next release will be out on May 14, and it will contain this fix.

**Comment by kjpou1** [comment]
So the 1.11.8 did not contain this fix?  Thank you for the information because I was in the process of creating a new issue against the release.

**Comment by MikeHolman** [comment]
Yeah, unfortunately fix went in a little too late to make 1.11.8.
